### DISCUSSION TITLE
Is there a Way to display JSON in pretty Print in Allure report

### DISCUSSION THREAD

**Discussion by deadshot674gam** [allure-framework/allure2#2007] [Questions & Support]
I have posted my question in StackOverflow - [LINK](https://stackoverflow.com/questions/75010377/how-can-i-add-json-diff-in-allure-report)

I have a useCase where two JSONs are compared and if there properties + values doesnt match then i want to pretty print the JSON diff OBject(Difference between the two JSONs) in a Indented and formatted Text not like crumbled text.

**Comment by artem-korolev** [comment]
We use Allure report as one of quality assurance tool in our integration platform. We do use newman adapter.

And our customers are also asking for this feature to have JSON/XML/etc request/response bodies to be pretty printed. At the current moment request/response bodies of the test are shown raw.

I understand why Allure decided to do like this. Show original request/response data makes total sense. But I think there must be a button to simply pretty print it.

So, if this functionality can be added to Allure, I'll be more then happy. If I know for sure Allure will accept this feature, I will start formalizing it, make feature request, and prepare pull request myself.
